app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,37 @@
+class PostsController < ApplicationController
+  def index
+  end
+
+  def show
+  end
+
+  def new
+    @category = Category.find(params[:category_id])
+    @post = Post.new
+  end
+
+  def edit
+  end
+
+  def create
+    @category = Category.find(params[:category_id])
+    @post = @category.posts.create(post_params)
+    
+    redirect_to category_path(@category)
+  end
+
+  def update
+    @category = Category.find(params[:category_id])
+    @post = @category.posts.find(params[:id])
+    @post.update(post_params)
+    redirect_to category_path(@category)
+  end
+
```

Comment:
this 'update' and it's associated 'new posts form' was a major challenge finally solved with Rahmal's help. It seems Rails often dictates the needed code expressed through error messages.